### ISSUE TITLE
[BUG] Unable to build on Raspberry Pi 3 using Debian stretch

### ISSUE THREAD

**Issue by dirkesquire** [Snapchat/KeyDB#525]
**Describe the bug**

I'm currently unable to build on Raspberry Pi 3 using Debian stretch.

**To reproduce**

The steps I followed was:
```
sudo apt-get update
sudo apt-get install build-essential nasm autotools-dev autoconf libjemalloc-dev tcl tcl-dev uuid-dev libcurl4-openssl-dev git
git clone https://github.com/EQ-Alpha/KeyDB.git
cd KeyDB
make
sudo make install
```

And the error is:

```
pi@Pi3:~/Builds/KeyDB $ make
cd src && make all
make[1]: Entering directory '/home/pi/Builds/KeyDB/src'
    CC adlist.o
    CC quicklist.o
    CC ae.o
    CC anet.o
    CC dict.o
    CC server.o
    CC sds.o
    CC zmalloc.o
    CC lzf_c.o
    CC lzf_d.o
    CC pqsort.o
    CC zipmap.o
    CC sha1.o
    CC ziplist.o
    CC release.o
    CC networking.o
    CC util.o
    CC object.o
object.cpp: In function ‘char* serializeStoredStringObject(sds, robj_roptr)’:
object.cpp:1585:5: error: static assertion failed
     static_assert((sizeof(robj) + sizeof(mvcc)) == sizeof(redisObjectStack), "");
     ^~~~~~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-atomic-alignment’
cc1plus: warning: unrecognized command line option ‘-Wno-address-of-packed-member’
Makefile:482: recipe for target 'object.o' failed
make[1]: *** [object.o] Error 1
make[1]: Leaving directory '/home/pi/Builds/KeyDB/src'
Makefile:6: recipe for target 'all' failed
make: *** [all] Error 2
pi@Pi3:~/Builds/KeyDB $ openssl version -a
OpenSSL 1.1.0l  10 Sep 2019
built on: reproducible build, date unspecified
platform: debian-armhf
options:  bn(64,32) rc4(char) des(long) blowfish(ptr)
compiler: gcc -DDSO_DLFCN -DHAVE_DLFCN_H -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/lib/ssl\"" -DENGINESDIR="\"/usr/lib/arm-linux-gnueabihf/engines-1.1\""
OPENSSLDIR: "/usr/lib/ssl"
ENGINESDIR: "/usr/lib/arm-linux-gnueabihf/engines-1.1"
```

**Expected behavior**

I would expect the build to succeed.

**Additional information**

Here is the code from object.cpp:

```
sds serializeStoredStringObject(sds str, robj_roptr o)
{
    uint64_t mvcc;
    mvcc = mvccFromObj(o);
    str = sdscatlen(str, &mvcc, sizeof(mvcc));
    str = sdscatlen(str, &(*o), sizeof(robj));
    static_assert((sizeof(robj) + sizeof(mvcc)) == sizeof(redisObjectStack), "");
    switch (o->encoding)
    {
    case OBJ_ENCODING_EMBSTR:
    case OBJ_ENCODING_RAW:
        str = sdscatsds(str, (sds)szFromObj(o));
        break;

    case OBJ_ENCODING_INT:
        break;  //nop
    }
        
    return str;
}
```

The only reference to "-Wno-atomic-alignment" I can find in the repository is in the makefile:

```
# Default settings
STD=-pedantic -DREDIS_STATIC=''
CXX_STD=-std=c++14 -pedantic -fno-rtti -D__STDC_FORMAT_MACROS  
ifneq (,$(findstring clang,$(CC)))
  STD+=-Wno-c11-extensions
else
ifneq (,$(findstring FreeBSD,$(uname_S)))
  STD+=-Wno-c11-extensions
endif
endif
WARN=-Wall -W -Wno-missing-field-initializers -Wno-address-of-packed-member -Wno-atomic-alignment
OPT=$(OPTIMIZATION)
```

**Comment by dirkesquire** [comment]
`uname -m`
> armv7l

Looks like it is indeed 32-bit architecture :(

**Comment by JohnSully** [comment]
Hey @dirkesquire its been a long time since I wrote that, but I believe your failing the assert due to padding and that it’s ok to comment it out. 

I can’t gurantee that using active replication will work on 32-bit as there’s some really nasty bit twiddling happening to save memory that hasn’t been validated there.

**Comment by dirkesquire** [comment]
Thank you so much that worked!

I commented out the static_assert in object.cpp and a two other places that was necessary (mostly in the replication areas), and it then compiled successfully!
However on running `make test` it wasn't happy and when starting the server it displayed the logo and seemed to start, but then complained about memory, and then halted.

So my second solution was to download the 64-bit of Raspberry Pi OS Lite. (https://www.raspberrypi.com/software/operating-systems/)
and it all just worked right away! This time I was able to install using APT `sudo apt install keydb` with the instructions under [KeyDB PPA Installation](https://docs.keydb.dev/docs/ppa-deb#keydb-ppa-installation)
